pkg/apis/componentconfig/types.go

```diff
@@ -554,3 +554,71 @@ const (
 	// "kube-scheduler" is the default scheduler lock object name
 	SchedulerDefaultLockObjectName = "kube-scheduler"
 )
+
```

Comment:
others are moving out of here. We need a plan where componentconfigs should go.

I still think something like k8s.io/componentconfig/kube-apiserver for the versioned types and cmd/kube-apiserver/.... for internal types makes most sense.

/cc @mtaufen @luxas @liggitt

Reply:
Why `cmd/kube-apiserver/...`, not `pkg/kubeapiserver/apis/apiserverconfig` like the kubelet for example?
`pkg` must not depend on `cmd` and `pkg` should depend on internal types, right?
`k8s.io/componentconfig/...` for the external types SGTM

Reply:
since the API groups are `*.config.k8s.io`, perhaps we should use `k8s.io/config/*` as the path convention when we publish to a staging repo?